Add azure storage account export target

cklingspor opened the pull request:
This PR prepares:

- `data_types`, `ignore_alloc_keys` & `rename_columns_config` to be read from ConfigMaps by reading them from file
- Business logic to use Service Principal and upload blob to Azure Storage Account container
- Factory pattern to target aws & azure blob storages. As of now, the code would not allow saving to local file system
- Testing extended accordingly
- Lots of thoughts what could be added to the code or where error handling could be improved
- Adjust README accordingly. But I wanted to checkin with you guys first.

mattray commented:
I'll have to defer to @lmello and/or @sntxrr on this one

lmello commented:
Thanks for your contribution. I am reviewing it.

lmello commented:
@cklingspor 

I love the idea of moving data_types, ignore_alloc_keys & rename_columns_config to config maps. This is what I was planning to do to make this more parametrized.  

I like the idea of supporting both s3 and azure blob storage, but must still support local filesystem. 

Please let me know when you are ready for review, Since I am using this in my production environment I will need to extensively test it before merging.

lmello commented on src/opencost_parquet_exporter.py:
```diff
@@ -69,89 +91,54 @@ def get_config(
         window_end = yesterday+'T23:59:59Z'
     window = f"{window_start},{window_end}"
+    config['window_start'] = window_start
     config['params'] = (
```
Can you also parametrize includeIdle, idleByNode ?

lmello commented on src/opencost_parquet_exporter.py:
```diff
@@ -69,89 +91,54 @@ def get_config(
         step = os.environ.get('OPENCOST_PARQUET_STEP', '1h')
+    if storage_backend is None:
+        storage_backend = os.environ.get(
+            'OPENCOST_PARQUET_STORAGE_BACKEND', 'aws')  # For backward compatibility
```
Thanks for keeping backward compatibility.

lmello commented on src/opencost_parquet_exporter.py:
```diff
@@ -69,89 +91,54 @@ def get_config(
         ("format", "json"),
-        ("step", step)
+        ("step", step),  # TODO: make this optional
+        ("accumulate", "true")  # TODO: Make this one a variable
```
This was not set before, it is better if we keep the previous behaviour and parametrize this in case you need to set it to true to your usecase.

lmello commented on src/opencost_parquet_exporter.py:
```diff
@@ -1,14 +1,31 @@
+from storage_factory import get_storage
+
+
+def load_config_file(file_path: str):
```
add some test cases related to the data types, ignore_alloc_keys and rename_columns

lmello reviewed (changes requested):
Thanks for the contribution, I really appreciate it. left some comments for you. 

let me know when you get it ready so I can test on my environment before merging.

cklingspor commented:
@lmello : I've added  the env variables (was using that already in a private version. I've also updated the docs. I think you could run a test on your environment. Interested to hear your findings 😊

lmello commented:
Sorry for the long delay. I am updating my opencost environment this week to version 1.110 and will be able to test this change next week. hope it is ok for you @cklingspor

cklingspor commented:
No worries, works for me @lmello . Thanks!

lmello commented:
@cklingspor sorry for the delay to test this. Trying to get some time at work. I hope to have a faster turn around on your next contributions.

lmello commented on src/opencost_parquet_exporter.py:
```diff
@@ -243,63 +268,53 @@ def save_result(processed_result, config):
     print("Starting run")
+    print("Load data types")
+    data_types = load_config_file(
+        file_path='./src/data_types.json')  # TODO: Make path ENV var
```
Please update the Dockerfile file to support this.  
The dockerfile need major update for this PR also because of the new files under src/ used for azure. 

Remember to also change the examples kubernetes resource if the entry point changes.

cklingspor replied:
Hi @lmello,
I've updated the PR. Was not 100% sure how you envisioned allowing for changes of `ENTRYPOINT` so please have look and let me know if you had something different in mind. I hope the changes address all requested changes.

lmello commented:
I liked that by adding the files we allow people to overwrite them on their docker images or using config maps.

cklingspor commented:
Hi @lmello , 
was on a longer vacation. Thank you for the review. I'll take a look the upcoming week and submit changes.

lmello commented:
Its good you added this, because I will need to deploy to gcp and you create the basis for supporting multiple providers, I am going to be adding the gcp support soon.

lmello commented:
Merging this PR! Thanks for your contribution @cklingspor !!!!
